Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Instance/InstanceUpdateExecutor.cpp

```diff
@@ -267,6 +268,36 @@ namespace AzToolsFramework
                             AzToolsFramework::EditorEntityContextRequestBus::Broadcast(
                                 &AzToolsFramework::EditorEntityContextRequests::HandleEntitiesAdded, newEntities);
 
+                            auto removeOrphanedEntities = [](Instance& instance)
+                            {
+                                instance.RemoveEntities(
+                                    [&instance](const AZStd::unique_ptr<AZ::Entity>& entity)
+                                    {
+                                        AZ::EntityId parentId;
+                                        if (AzToolsFramework::Components::TransformComponent* transformComponent =
+                                            entity->FindComponent<AzToolsFramework::Components::TransformComponent>())
+                                        {
+                                            parentId = transformComponent->GetParentId();
+                                        }
+
+                                        return (!parentId.IsValid() || !instance.GetEntityAlias(parentId).has_value());
+                                    });
+                            };
+
+                            // Remove any orphaned entities from the instance object (patch data stays intact).
+                            // Entities can contain unregistered parent aliases in the target template after
+                            // applying add/remove entity patches to the source template. These show up as
+                            // invalid parent EntityIds.
+                            // Note: This needs to be validated after the HandleEntitiesAdded call to ensure
+                            // parent entityIds have been set up (ex. when creating a new entity, its parent
+                            // is set up on a callback triggered via HandleEntitiesAdded)
+                            removeOrphanedEntities(*instanceToUpdate);
```

Comment:
Rather than do it after loading the prefab, we can do this during the loading of the transform component itself : https://github.com/o3de/o3de/blob/development/Code/Framework/AzToolsFramework/AzToolsFramework/ToolsComponents/TransformComponentSerializer.cpp#L39-L43

In that code, if m_parentEntityId is invalid, we can mark the entity and remove it from the instance in InstanceSerializer.cpp or EntitySerializer.cpp

This will ensure that we don't need to check the parent of every entity in a prefab instance recursively(which can be a lot) and won't impact performance.

Reply:
When the transform component is loaded, we are not guaranteed that its parent entity is already loaded yet.

Reply:
oh, that's right. The entity id is actually valid. However, the instance won't have an entity with that id. But we can still do the same checks you are doing here in InstanceSerializer.cpp this way:

```
Load Instance()
{
	// Create a new struct to store a vector of transform component parents and add it to JsonSerializationMetadata
	while loading each entity()
	{
		LoadTransformComponent()
		{
			add parent id to vector.
		}
	}
	
	for each (parentId : vector)
	{
		if (parentId is not present in Instance)
		{
			// Remove from instance
		}
	}
}
```

If we do it this way, we can avoid 

1. The additional overhead to recurse over every nested instance twice (once during loading and once after loading)
2. The time to find the transform component in each entity (O(n) for each entity, where n is number of components in entity)

Reply:
Could you explain #1 in further detail? This logic should happen whenever an instance needs to be updated due to propagation.

Reply:
The add/remove entity patches still exist despite the entity not being part of the instance. When template propagation occurs, it reapplies the patches on the target and so we need to recheck this logic.